### PR TITLE
chore: resolve lint warnings - < Menu />

### PR DESCRIPTION
**Pull request by Kritik-J**
fixes: #1648

### CODE REVIEW

File: src/components/Menu/MenuItemButton/MenuItemButton.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ import styles from "./MenuItemButton.module.scss";
 interface MenuItemButtonProps extends VibeComponentProps {
   /** Backward compatibility for props naming **/
   classname?: string;
-  kind?: typeof MenuItemButton.kinds[keyof typeof MenuItemButton.kinds];
+  kind?: (typeof MenuItemButton.kinds)[keyof typeof MenuItemButton.kinds];
```
Comment (SergeyRoyt) [inline src/components/Menu/MenuItemButton/MenuItemButton.tsx:19]: Lint issue

File: src/components/Menu/Menu/Menu.tsx [hunk 2]
```diff
@@ -34,7 +34,7 @@ import styles from "./Menu.module.scss";
 export interface MenuProps extends VibeComponentProps {
   /** Backward compatibility for props naming **/
   classname?: string;
-  size?: typeof SIZES[keyof typeof SIZES];
+  size?: (typeof SIZES)[keyof typeof SIZES];
```
Comment (SergeyRoyt) [inline src/components/Menu/Menu/Menu.tsx:37]: Lint issue

File: src/components/Menu/Menu/Menu.tsx [hunk 3]
```diff
@@ -87,15 +87,20 @@ const Menu: VibeComponent<MenuProps> & {
 
     const children = useMemo(() => {
       const allChildren = React.Children.toArray(originalChildren);
```
Comment (SergeyRoyt) [inline src/components/Menu/Menu/Menu.tsx:89]: I propose we cast the type here for all the array, like this
```suggestion
      const allChildren = React.Children.toArray(originalChildren) as MenuChild[];
```
And let's define MenuChild in `MenuConstants` and reuse it, also let's make the type field optional to avoid the errors, should be smth like this:
```
type MenuChild = ReactElement & {
  type?: Record<string, unknown>;
};
```

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
Wrote a few comments. Let's make sure that all the checks are green - test and lint

File: src/components/Menu/Menu/utils/utils.ts [hunk 4]
```diff
@@ -1,7 +1,10 @@
 import { ReactElement } from "react";
 
-export function isMenuChildSelectable(child: ReactElement): boolean {
-  // @ts-ignore
+export function isMenuChildSelectable(
+  child: ReactElement & {
+    type: Record<string, unknown>;
+  }
```
Comment (SergeyRoyt) [inline src/components/Menu/Menu/utils/utils.ts:6]:
```suggestion
  child: MenuChild;
```
+ lint fixes

File: src/components/Menu/MenuItem/MenuItem.tsx [hunk 5]
```diff
@@ -117,9 +117,12 @@ const MenuItem: VibeComponent<MenuItemProps> & {
     const isSubMenuOpen = !!children && isActive && hasOpenSubMenu;
     const hasChildren = !!children;
     const shouldShowSubMenu = hasChildren && isParentMenuVisible && isSubMenuOpen;
-    const submenuChild = children && React.Children.only(children);
+    const submenuChild =
+      children &&
+      (React.Children.only(children) as ReactElement & {
+        type: Record<string, unknown>;
+      });
```
Comment (SergeyRoyt) [inline src/components/Menu/MenuItem/MenuItem.tsx:124]: Let's revert this change and instead change type of `children` prop:
```diff
-   children?: ReactElement | ReactElement[];
+   children?: MenuChild | MenuChild[];
```

Comment (SergeyRoyt) [reply src/components/Menu/MenuItem/MenuItem.tsx:124]:
```suggestion
    const submenuChild: MenuChild = children && React.Children.only(children);
```
Cause still React.ChildrenOnly will return ReactElement

**Comment by Kritik-J** [comment]
@SergeyRoyt please review

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
Looks good, but some tests are still not green, let's try to fix them 🙏

**Comment by Kritik-J** [comment]
@SergeyRoyt all checks are green now

**Review by SergeyRoyt — APPROVED** [review]
Nice! Thanks a lot! 🙂
